doc/rm/comportability_specification/index.md

```diff
@@ -440,18 +439,71 @@ Unlike interrupts, there is no software component to alerts at the peripheral, t
+
+1. *Recoverable*, one-time triggered alerts.
+This category is for regular alerts that are due to recoverable error conditions.
+The alert sender transmits one single alert event when the corresponding error condition is asserted.
+
+2. *Fatal* alerts that are continuously triggered until reset.
+This category is for highly critical alerts that are due to terminal error conditions.
+The alert sender continuously transmits alert events until the system is reset.
+
+It is recommended that fatal alerts also trigger local security countermeasures, if they exist.
+For example, a redundantly encoded FSM that is glitched into an invalid state is typically considered to be a fatal error condition.
+In this case, a local countermeasure could be to move the FSM into a terminal error state in order to render the FSM inoperable until the next reset.
+
+The table below lists a few common error conditions and the recommended alert type for each of those errors.
+
+Error Event                                                             | Regular IRQ | Recoverable Alert | Fatal Alert
+------------------------------------------------------------------------|-------------|-------------------|-------------
+ECC correctable in NVM (OTP, Flash)                                     | (x)         | x                 |
+ECC uncorrectable in Flash                                              | (x)         | x                 |
+ECC uncorrectable in OTP                                                | (x)         |                   | x
+Any ECC / Parity error in SRAMs or register files                       | (x)         |                   | x
+Glitch detectors (e.g., invalid FSM encoding)                           | (x)         |                   | x
+Incorrect usage of security IP (e.g., shadowed control register in AES) | (x)         | x                 |
+Incorrect usage of regular IP                                           | x           |                   |
+
+(x): optional
+
+The column "Regular IRQ" indicates whether the corresponding error condition should also send out a regular IRQ.
+A peripheral may optionally send out an IRQ for any alert event, depending on whether this is needed by the programming model to make forward progress.
+Note that while alerts may eventually lead to a system wide reset, this is not guaranteed since the alert response depends on the alert handler configuration.
+
+### Defining Alerts
+
+The Hjson configuration file defined above specifies all that needs to be known about the alerts in the standard case.
+The following sections specify what comes out of various tools based upon the simple list defined in the above example.
+
+In terms of naming convention, alerts shall be given a meaningful name that is indicative of its cause.
+Recoverable alerts must be prefixed with `recov_*`, whereas fatal alerts must be prefixed with `fatal_*`.
+For instance, an uncorrectable parity error in SRAM could be named `fatal_parity_error`.
```

Comment:
Did you remove the case for a single alert of each type intentionally? Or should we always have a prefix and (more than one) fine-grained alerts?

Reply:
I removed it since @sriyerg suggested it would be good to stay on the more "descriptive" side even in that case.

Reply:
That however does not mean you necessarily need to have more than one alert.
Just the name should be descriptive in any case.

Reply:
I am surprised to see this case being removed as I thought there was consensus that we wanted to allow both and I've also adapted aes accordingly, already. For the majority of the simpler peripherals we will have in the system (e.g. aes, hmac, not tied up with other system-level blocks like otp, lc, keymgr), a single alert per type will be enough and it would help to avoid the issues we are now seeing around interrupts (see #3421). By the way, the latest guidance is to consolidate multiple interrupts in new designs where possible.

In addition, if we want meaningful and descriptive names it basically means that whenever we hook up another alert source in the process of security hardening, this implies changes at the top-level as well.

Reply:
Sorry, I may have moved a bit too fast here.

Let me clarify: this was not meant to discourage consolidating multiple alert sources into a single alert. Rather, it was meant to encourage more descriptive naming of the alerts if possible.

I do however see that in some cases where many different alert sources are bundled into a single alert, finding a descriptive name can be difficult, and we may end up just giving the alert a very generic sounding extension (like `fatal_event`, which does not convey any more information than `fatal`).

So how about we modify this slightly to still encourage descriptive naming, if that is possible, but allow to fall back to `fatal` and `recov` for bundled alerts with many different alert sources?

Reply:
This sounds good to me Michael :+1: